### PR TITLE
Expose new variants of EventCounter APIs

### PR DESCRIPTION
**Pull request by sywhang**
Exposing the new EventCounter variant APIs in reference assembly. 

API review: https://github.com/dotnet/corefx/issues/36129

### CODE REVIEW

File: src/System.Diagnostics.Tracing/ref/System.Diagnostics.Tracing.cs [hunk 1]
```diff
@@ -53,11 +53,36 @@ public partial class EventCommandEventArgs : System.EventArgs
...
+    {
+        public PollingCounter(string name, System.Diagnostics.Tracing.EventSource eventSource, Func<double> metricProvider) : base(name, eventSource) { }
+    }
+    public partial class IncrementingEventCounter : System.Diagnostics.Tracing.DiagnosticCounter
+    {
+        public IncrementingEventCounter(string name, System.Diagnostics.Tracing.EventSource eventSource) : base(name, eventSource) { }
+        public void Increment(double increment = 1) { }
+        public TimeSpan DisplayRateTimeScale { get { throw null; } set { } }
+    }
+    public partial class IncrementingPollingCounter : System.Diagnostics.Tracing.DiagnosticCounter
```
Comment (stephentoub) [inline src/System.Diagnostics.Tracing/ref/System.Diagnostics.Tracing.cs]: Are these the type names agreed to in the API review? Looking at it here, it's strange to me that IncrementingEventCounter didn't derive from EventCounter, and the same for {IncrementingPollingCounter}.

File: src/System.Diagnostics.Tracing/ref/System.Diagnostics.Tracing.cs [hunk 2]
```diff
@@ -53,11 +53,36 @@ public partial class EventCommandEventArgs : System.EventArgs
...
+    {
+        public IncrementingPollingCounter(string name, System.Diagnostics.Tracing.EventSource eventSource, Func<double> totalValueProvider) : base(name, eventSource) { }
+        public TimeSpan DisplayRateTimeScale { get { throw null; } set { } }
```
Comment (stephentoub) [inline src/System.Diagnostics.Tracing/ref/System.Diagnostics.Tracing.cs]: What about tests?

File: src/System.Diagnostics.Tracing/ref/System.Diagnostics.Tracing.cs (hunk 1 above)
Comment (sywhang) [reply src/System.Diagnostics.Tracing/ref/System.Diagnostics.Tracing.cs]: Yes, the names were agreed to during the API review meeting. 
`EventCounter` now derives from a base class `DiagnosticCounter`, and all these other counters do as well.

File: src/System.Diagnostics.Tracing/ref/System.Diagnostics.Tracing.cs (hunk 2 above)
Comment (sywhang) [reply src/System.Diagnostics.Tracing/ref/System.Diagnostics.Tracing.cs]: There are some tests that are checked in CoreCLR: https://github.com/dotnet/coreclr/tree/master/tests/src/tracing/eventcounter. I've tested these end-to-end scenarios using dotnet-counters (https://github.com/dotnet/diagnostics/tree/master/src/Tools/dotnet-counters), but I'll be checking in smaller tests to CoreCLR soon.

**Comment by sywhang** [comment]
I'm splitting the implementation of EventCounter in the ref assembly because of #36873.

**Comment by jkotas** [comment]
> I'm splitting the implementation of EventCounter in the ref assembly because of #36873.

@sywhang It is better to just add ApiCompatBaseline like this: https://github.com/dotnet/corefx/blob/master/src/System.Runtime/src/ApiCompatBaseline.uapaot.txt . It is what we do to workaround the synchronization delay with .NET Native.

**Comment by jkotas** [comment]
Also, please "squash and merge" next time and use appropriate commit descriptions. Your "I'm stupid" commit description is recorded in the git history forever now.

**Comment by ahsonkhan** [comment]
Yes, please (squash and merge). I just saw this in the git log and was about to mention the same thing!

As an aside, consider more meaningful commit messages as well, just in case. :)

**Pull request merged**